src/journal/JournalRecorder.h

```diff
@@ -38,6 +38,7 @@ class JournalRecorder {
 
 private:
   typedef std::map<uint8_t, ObjectRecorderPtr> ObjectRecorderPtrs;
+  typedef std::vector<std::unique_ptr<Mutex::Locker>> Lockers;
```

Comment:
This ends up being `unique_ptr<lock_guard<Mutex>>`, correct?  Could `unique_lock<Mutex>` be used instead?

Reply:
I was not able to make it compile. And I did not find examples of `unique_lock<Mutex>`  usage in that branch.